include/rmm/mr/device/per_device_resource.hpp

```diff
@@ -90,14 +90,20 @@ namespace mr {
 
 namespace detail {
 
+// These symbols must have default visibility so that when they are
+// referenced in multiple different DSOs the linker correctly
+// determines that there is only a single unique reference to the
+// function symbols (and hence they return unique static references
+// across different DSOs).
```

Comment:
Perhaps also explain that while this is subsumed by `RMM_NAMESPACE`, we want these functions explicitly annotated in case the namespace definition ever changes?

Reply:
Ah, this one slipped through the cracks. Now done.